src/app/page/admin/decks/admin.decks.service.ts

```diff
@@ -23,30 +26,38 @@ export class AdminDecksService {
         DTOConverter.jsonArrayToCollection(DTOConverter.jsonToAdminDeck, response.json()))
       .catch(this.handleError);
   }
-
-  deleteDeck(deckId: number) {
-    return this.orlp.delete('/api/admin/decks/' + deckId);
-}
+  deleteDeck(link: Link) {
+    let shortLink: string = this.orlp.getShortLink(link);
+    shortLink = this.orlp.decodeLink(shortLink);
```

Comment:
It feels like you are initializing one variable twice, I would either combine those two lines in one, either use another one on the second line.

Reply:
yes it's true, thank you.